agent/php_internal_instrument.c

```diff
@@ -1531,6 +1533,71 @@ NR_INNER_WRAPPER(memcache_function) {
                                              INTERNAL_FUNCTION_PARAM_PASSTHRU);
 }
 
+NR_INNER_WRAPPER(memcached_add_server) {
+  char* host = NULL;
+  nr_string_len_t host_len = 0;
+  zend_long port = 0;
+  zend_long weight = 0;
+  nr_datastore_instance_t* instance = NULL;
+  char* instance_metric = NULL;
+  int zcaught = 0;
+
+  if (SUCCESS
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS(), "s|ll", &host,
+          &host_len, &port, &weight) &&
+      NULL != host) {
+    instance = nr_php_memcached_create_datastore_instance(host, port);
+    instance_metric = nr_formatf("Datastore/instance/Memcached/%s/%s",
+                               instance->host, instance->port_path_or_id);
+    nrm_force_add(NRPRG(txn)->unscoped_metrics, instance_metric, 0);
+    nr_free(instance);
```

Comment:
[nr_free](https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/util_memory.h#L159) only frees the instance pointer, it doesn't free the strduped fields it was pointing to.

This needs [nr_datastore_instance_destroy](https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/nr_datastore_instance.c#L37) to nicely cleanup all the strings that were strduped.

Reply:
addressed in [18fff7a](https://github.com/newrelic/newrelic-php-agent/pull/958/commits/18fff7abc952385e4eea2925f631906fc7463e3b) and [c40cbd4](https://github.com/newrelic/newrelic-php-agent/pull/958/commits/c40cbd42f74ca295924f55fbd6a9deab996b2085)